config/management/genesis/src/command.rs

```diff
@@ -98,73 +98,47 @@ impl Command {
     }
 
     pub fn create_and_insert_waypoint(self) -> Result<Waypoint, Error> {
-        match self {
-            Command::CreateAndInsertWaypoint(cmd) => cmd.execute(),
-            _ => Err(self.unexpected_command(CommandName::CreateAndInsertWaypoint)),
-        }
+        execute_command!(
+            self,
+            Command::CreateAndInsertWaypoint,
+            CommandName::CreateAndInsertWaypoint
+        )
     }
 
     pub fn create_waypoint(self) -> Result<Waypoint, Error> {
-        match self {
-            Command::CreateWaypoint(cmd) => cmd.execute(),
-            _ => Err(self.unexpected_command(CommandName::CreateWaypoint)),
-        }
+        execute_command!(self, Command::CreateWaypoint, CommandName::CreateWaypoint)
```

Comment:
do you need both arguments? there's a `impl From<&Command> for CommandName`

Reply:
We actually do need both arguments.  The CommandName is the expected command

Reply:
I've gone ahead and renamed it accordingly though to be less confusing.

Reply:
the second and third argument are always the command and its corresponding name, no?

Reply:
Yeah, but the problem comes more with the way that macros are defined.

In this case the `path` `Command::CreateWaypoint` refers to the path of the type, but not an instantiation of the object, which would include the `Command::CreateWaypoint(cmd)`, where the `cmd` needs to be defined.  So, sadly `CommandName::from(Command::CreateWaypoint)` doesn't work, because it's not an actual instantiation of the enum. :(

Reply:
If we find a better way we can come back and fix it later :)